### DISCUSSION TITLE
pwsh completion

### DISCUSSION THREAD

**Discussion by silverqx** [composer/composer#10882] [Q&A / Support] [answered]
Do you know about any pwsh completion script for composer?

**Comment by Seldaek** [comment, marked as answer]
Composer 2.4-dev adds support for bash completions using https://symfony.com/blog/new-in-symfony-5-4-console-autocompletion - you can try it with `composer self-update --snapshot` then `composer completion bash`.

I don't know how powershell completion works, perhaps you could adapt a solution from this, or perhaps it'd require support for it in upstream symfony components which would make it way more complex unfortunately.

**Comment by silverqx** [comment]
I know about the bash completion, I'm using it on Linux.

pwsh completion is simple, I have created the pwsh completion for my library a few days/week ago, and it can be practically summarized in a few lines on how it works.

If you are interested in I can describe it, it's a few days of work to add support for it, if not then nothing happens, everybody has too much work still, but it would be nice to have pwsh completions for the composer.

**Reply by Seldaek** [reply]
If you can work on an implementation for it I would say why not, but I personally won't have time to work on this for a long time I think, tons of other stuff which have higher priority.

**Comment by silverqx** [comment]
I'm in the same situation 😁 Last two weeks I have worked on bash/zsh/pwsh completions for my library and also have too much todo on personal projects 😎, its classic, thx for replies.

**Comment by dkarlovi** [comment]
You should upstream the Powershell completion support into Symfony console which will then trickle down into Composer (assuming Console 6.2 is allowed by Composer, which it should be?)

See this PR as an example of adding a new shell into Console: https://github.com/symfony/symfony/pull/47018

**Reply by Seldaek** [reply]
It's definitely something to add in symfony/console, but trickle down is a bit tricky because 6.1+ requires PHP 8.1, which we definitely cannot require. We can see if it's worth down-compiling code when we have a compelling reason to upgrade. For now I'm still really happy we're on 5.4 and not stuck on 2.8 anymore :D

**Reply by dkarlovi** [reply]
Not require, but **allow**. This could mean I get to construct a PHP 8.1, Symfony 6.2 Console version of Composer which supports Powershell etc.

**Reply by GromNaN** [reply]
We would need 2 compiled versions of composer.phar: one for PHP 7.2 with Sf 5.4, the other for PHP 8.1 with Sf 6.2+).

**Reply by dkarlovi** [reply]
That doesn't sound too difficult to set up and could be a nice opportunity for Composer to keep getting new features from Symfony, while still providing BC for users who need it. :+1:
